provision/Dockerfile

```diff
@@ -1,20 +1,43 @@
-FROM ubuntu:latest
+FROM debian:bookworm-slim AS build
 
-RUN apt-get update
+WORKDIR /build
 
-RUN apt-get -y install wget
-
-WORKDIR /usr/src
+RUN apt-get update \
+&& apt-get install -y wget \
+&& apt-get clean \
+&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 RUN wget https://github.com/dogecoin/dogecoin/releases/download/v1.14.5/dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
-RUN tar -xzvf dogecoin-1.14.5-x86_64-linux-gnu.tar.gz
-RUN chmod +x dogecoin-1.14.5/bin/dogecoind dogecoin-1.14.5/bin/dogecoin-cli
-RUN ln -s /usr/src/dogecoin-1.14.5/bin/dogecoind /usr/bin/dogecoind
-RUN ln -s /usr/src/dogecoin-1.14.5/bin/dogecoin-cli /usr/bin/dogecoin-cli
+RUN tar -xzf dogecoin-1.14.5-x86_64-linux-gnu.tar.gz \
+    --wildcards "*/dogecoin*" \
+    --exclude="*-qt*" \
+    --strip-components=1 \
+    --one-top-level=dogecoin
+
+FROM debian:bookworm-slim
+
+RUN useradd dogecoin \
+    && apt-get update && apt-get install gosu \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+ENV DOGECOIN_DATA=/root/.dogecoin
+
+COPY --from=build --chown=dogecoin:dogecoin /build/dogecoin/bin/dogecoin* /usr/local/bin/
+
+COPY dogecoin.conf ${DOGECOIN_DATA}/dogecoin.conf
+
+COPY entrypoint.sh /entrypoint.sh
+
+RUN ["chmod", "+x", "/entrypoint.sh"]
+
+VOLUME ["/dogecoin/.dogecoin"]
+
+WORKDIR '/dogecoin'
 
-ADD dogecoin.conf /root/.dogecoin/dogecoin.conf
+EXPOSE 22555 22556 44555 44556 18880
```

Comment:
Did the ports changed for regtest in version 1.14.5 ?

Reply:
no i was doing testing on my local with 2 nodes so i reserved regular regtest port for your spv node.